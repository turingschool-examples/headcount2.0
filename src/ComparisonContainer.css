
.comp-container {
  padding-top: 2rem;
  background-color: #C2B2B4;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

.comp-card {
  padding: .5rem;
  border: 2px solid chartreuse;
  height: 20rem;
  width: 15vw;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  background-color: #ffffff;
  border-radius: .9rem;
}

.total-card {
  padding: .5rem;
  height: 15rem;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  border: 2px solid chartreuse;
  width: 15vw;
  background-color: #ffffff;

}