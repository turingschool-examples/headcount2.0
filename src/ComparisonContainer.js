import React from 'react';
import Card from './Card';

const ComparisonContainer = () => {

  return(
    <div>
      <h3>Select two districts to compare, they will be displayed here.</h3>
    </div>
  )
}

export default ComparisonContainer;