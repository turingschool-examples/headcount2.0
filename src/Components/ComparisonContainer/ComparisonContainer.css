.comparison-container {
  display: flex;
  justify-content: space-evenly;
  flex-direction: row;
  flex-wrap: wrap;
  max-width: 1200px;
  margin: auto
}

.comparison-card {
  font-family: avenir, sans-serif;
  font-size: 1em;
  text-align: center;
}

.district-average {
  font-size: 1.5em;
}

.comparison-figure {
  font-size: 3em;
  color: red;
}