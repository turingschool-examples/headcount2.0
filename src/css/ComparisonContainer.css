.comparison-container {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
}

@media only screen and (max-width: 1350px) {
  .comparison-container {
    grid-template-columns: repeat(3, 1fr) !important;
  }
}
