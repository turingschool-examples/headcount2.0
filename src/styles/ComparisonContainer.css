.comparison-cards {
  width:100%;
  height: 575px;
  background-color: lightblue;
  text-align: center;
}