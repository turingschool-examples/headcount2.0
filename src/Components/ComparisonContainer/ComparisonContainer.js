import React from 'react';
import './ComparisonContainer.css';
import DistrictCard from '../DistrictCard/DistrictCard'
import PropTypes from 'prop-types';


const ComparisonContainer = ({ comparedDistricts, selectDistrict, compareDistrictAverages, findAverage }) => {
  const districtNames = Object.keys(comparedDistricts)
  const districtCards = districtNames.map(district => {
    return <DistrictCard 
              key={comparedDistricts[district].key}
              district={comparedDistricts[district]} 
              selectDistrict={selectDistrict}
            />
  })

  if (districtNames.length === 2 ) {
    const comparison = compareDistrictAverages(districtNames[0], districtNames[1]);

    return (
      <div className='comparison-container'>
      { districtCards }
        <div className='comparison-card'>
          <h2>{districtNames[0]}:</h2>
          <p className='district-average'>{findAverage(districtNames[0])}</p>
          <h3 className='comparison-figure'>{comparison.compared}</h3>
          <p className='district-average'>{findAverage(districtNames[1])}</p>
          <h2>{districtNames[1]}:</h2>
        </div>
      </div>
    )

  }
    return (
      <div className="comparison-container">
        { districtCards }
      </div>
    )
} 


ComparisonContainer.propTypes = {
  comparedDistricts: PropTypes.object.isRequired
}

export default ComparisonContainer;