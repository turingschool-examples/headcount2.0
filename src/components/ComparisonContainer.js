import React, { Component } from 'react';
import Card from './Card';
import '../styles/ComparisonContainer.css';

const ComparisonContainer = (props) => {

  const dataArray = Object.keys(props.cardCompData)
  const newCard = dataArray.map((location, index) => <Card data={ props.cardCompData[location]} key={index}/>)

  return (
    <section className='comparison-cards'>
      <h1>Comparison Section</h1>
      {newCard}
    </section>
  )
}

export default ComparisonContainer