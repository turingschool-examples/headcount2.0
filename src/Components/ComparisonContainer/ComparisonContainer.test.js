import React from 'react';
import { shallow } from 'enzyme';
import ComparisonContainer from './ComparisonContainer';

describe('ComparisonContainer unit test', () => {
  let wrapper;
  let mockComparedDistricts; 
  let mockcompareDistrictAverages;
  let mockselectDistrict;
  let mockFindAverage;

  beforeEach(() => {
    mockComparedDistricts = { COLORADO: 
                              { stats: 
                                  { 2004: 0.24,
                                    2005: 0.278,
                                    2006: 0.337,
                                    2007: 0.395,
                                    2008: 0.536,
                                    2009: 0.598,
                                    2010: 0.64,
                                    2011: 0.672,
                                    2012: 0.695,
                                    2013: 0.703,
                                    2014: 0.741 },
                                location: 'COLORADO' },
                             ACADEMY20: 
                              { stats: 
                                  { 2004: 0.302,
                                    2005: 0.267,
                                    2006: 0.354,
                                    2007: 0.392,
                                    2008: 0.385,
                                    2009: 0.39,
                                    2010: 0.436,
                                    2011: 0.489,
                                    2012: 0.479,
                                    2013: 0.488,
                                    2014: 0.49},
                                location: 'ACADEMY 20' }
                            }
    mockselectDistrict = jest.fn();
    mockFindAverage = jest.fn();
    mockcompareDistrictAverages = jest.fn().mockReturnValue({comparison: {compared: 1}});                        
    wrapper = shallow(<ComparisonContainer 
                              comparedDistricts={mockComparedDistricts} 
                              selectDistrict={mockselectDistrict}
                              compareDistrictAverages={mockcompareDistrictAverages}
                              findAverage={mockFindAverage}
                            />)

  })
  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
    });

  it('should conditionally render district comparison information', () => {
    const actualState = wrapper.find('.comparison-card').length;
    const expectedState = 1
    expect(actualState).toEqual(expectedState);
    })
});
