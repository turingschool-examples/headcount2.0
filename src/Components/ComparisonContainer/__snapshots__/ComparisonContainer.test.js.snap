// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonContainer unit test should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    compareDistrictAverages={[Function]}
    comparedDistricts={
        Object {
            "ACADEMY20": Object {
              "location": "ACADEMY 20",
              "stats": Object {
                "2004": 0.302,
                "2005": 0.267,
                "2006": 0.354,
                "2007": 0.392,
                "2008": 0.385,
                "2009": 0.39,
                "2010": 0.436,
                "2011": 0.489,
                "2012": 0.479,
                "2013": 0.488,
                "2014": 0.49,
              },
            },
            "COLORADO": Object {
              "location": "COLORADO",
              "stats": Object {
                "2004": 0.24,
                "2005": 0.278,
                "2006": 0.337,
                "2007": 0.395,
                "2008": 0.536,
                "2009": 0.598,
                "2010": 0.64,
                "2011": 0.672,
                "2012": 0.695,
                "2013": 0.703,
                "2014": 0.741,
              },
            },
          }
    }
    selectDistrict={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <DistrictCard
            district={
                        Object {
                                    "location": "COLORADO",
                                    "stats": Object {
                                      "2004": 0.24,
                                      "2005": 0.278,
                                      "2006": 0.337,
                                      "2007": 0.395,
                                      "2008": 0.536,
                                      "2009": 0.598,
                                      "2010": 0.64,
                                      "2011": 0.672,
                                      "2012": 0.695,
                                      "2013": 0.703,
                                      "2014": 0.741,
                                    },
                                  }
            }
            selectDistrict={[Function]}
/>,
          <DistrictCard
            district={
                        Object {
                                    "location": "ACADEMY 20",
                                    "stats": Object {
                                      "2004": 0.302,
                                      "2005": 0.267,
                                      "2006": 0.354,
                                      "2007": 0.392,
                                      "2008": 0.385,
                                      "2009": 0.39,
                                      "2010": 0.436,
                                      "2011": 0.489,
                                      "2012": 0.479,
                                      "2013": 0.488,
                                      "2014": 0.49,
                                    },
                                  }
            }
            selectDistrict={[Function]}
/>,
        ],
        <div
          className="comparison-card"
>
          <h2>
                    COLORADO
          </h2>
          <h3 />
          <h2>
                    ACADEMY20
          </h2>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "district": Object {
            "location": "COLORADO",
            "stats": Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
              "2008": 0.536,
              "2009": 0.598,
              "2010": 0.64,
              "2011": 0.672,
              "2012": 0.695,
              "2013": 0.703,
              "2014": 0.741,
            },
          },
          "selectDistrict": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "district": Object {
            "location": "ACADEMY 20",
            "stats": Object {
              "2004": 0.302,
              "2005": 0.267,
              "2006": 0.354,
              "2007": 0.392,
              "2008": 0.385,
              "2009": 0.39,
              "2010": 0.436,
              "2011": 0.489,
              "2012": 0.479,
              "2013": 0.488,
              "2014": 0.49,
            },
          },
          "selectDistrict": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              COLORADO
</h2>,
            <h3 />,
            <h2>
              ACADEMY20
</h2>,
          ],
          "className": "comparison-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "COLORADO",
            },
            "ref": null,
            "rendered": "COLORADO",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ACADEMY20",
            },
            "ref": null,
            "rendered": "ACADEMY20",
            "type": "h2",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <DistrictCard
              district={
                            Object {
                                          "location": "COLORADO",
                                          "stats": Object {
                                            "2004": 0.24,
                                            "2005": 0.278,
                                            "2006": 0.337,
                                            "2007": 0.395,
                                            "2008": 0.536,
                                            "2009": 0.598,
                                            "2010": 0.64,
                                            "2011": 0.672,
                                            "2012": 0.695,
                                            "2013": 0.703,
                                            "2014": 0.741,
                                          },
                                        }
              }
              selectDistrict={[Function]}
/>,
            <DistrictCard
              district={
                            Object {
                                          "location": "ACADEMY 20",
                                          "stats": Object {
                                            "2004": 0.302,
                                            "2005": 0.267,
                                            "2006": 0.354,
                                            "2007": 0.392,
                                            "2008": 0.385,
                                            "2009": 0.39,
                                            "2010": 0.436,
                                            "2011": 0.489,
                                            "2012": 0.479,
                                            "2013": 0.488,
                                            "2014": 0.49,
                                          },
                                        }
              }
              selectDistrict={[Function]}
/>,
          ],
          <div
            className="comparison-card"
>
            <h2>
                        COLORADO
            </h2>
            <h3 />
            <h2>
                        ACADEMY20
            </h2>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "district": Object {
              "location": "COLORADO",
              "stats": Object {
                "2004": 0.24,
                "2005": 0.278,
                "2006": 0.337,
                "2007": 0.395,
                "2008": 0.536,
                "2009": 0.598,
                "2010": 0.64,
                "2011": 0.672,
                "2012": 0.695,
                "2013": 0.703,
                "2014": 0.741,
              },
            },
            "selectDistrict": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "district": Object {
              "location": "ACADEMY 20",
              "stats": Object {
                "2004": 0.302,
                "2005": 0.267,
                "2006": 0.354,
                "2007": 0.392,
                "2008": 0.385,
                "2009": 0.39,
                "2010": 0.436,
                "2011": 0.489,
                "2012": 0.479,
                "2013": 0.488,
                "2014": 0.49,
              },
            },
            "selectDistrict": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                COLORADO
</h2>,
              <h3 />,
              <h2>
                ACADEMY20
</h2>,
            ],
            "className": "comparison-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "COLORADO",
              },
              "ref": null,
              "rendered": "COLORADO",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ACADEMY20",
              },
              "ref": null,
              "rendered": "ACADEMY20",
              "type": "h2",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
